distributed/semaphore.py

```diff
@@ -460,22 +462,46 @@ def acquire(self, timeout=None):
         timeout = parse_timedelta(timeout)
         return self.client.sync(self._acquire, timeout=timeout)
 
+    async def _release(self):
+        # popleft to release the oldest lease first
+        lease_id = self._leases.popleft()
+        logger.info("%s releases %s for %s", self.client.id, lease_id, self.name)
+
+        try:
+            await retry_operation(
+                self.client.scheduler.semaphore_release,
+                name=self.name,
+                lease_id=lease_id,
+                operation="semaphore release: client=%s, lease_id=%s, name=%s"
+                % (self.client.id, lease_id, self.name),
+            )
+            return True
+        except Exception:  # Release fails for whatever reason
+            logger.error(
+                "Release failed for client=%s, lease_id=%s, name=%s. Cluster network might be unstable?"
+                % (self.client.id, lease_id, self.name),
+                exc_info=True,
+            )
+            return False
+
     def release(self):
         """
-        Release a semaphore.
+        Release the semaphore.
 
-        Increment the internal counter by one.
-        """
+        This will increment the amount of available leases by one.
```

Comment:
This is not necessarily true. This is true for the stdlib semaphore but not this one. the exception is that we cannot increase the number of available leases to a larger value than initially specified. This is also documented in the class documentation but I'm afraid this might be slightly misleading

Reply:
Can you suggest a better alternative?

Reply:
I've just removed that line for now, I don't think the previous line was very informative either